pkg/graveler/graveler.go

```diff
@@ -1903,16 +1905,135 @@ func (g *KVGraveler) Reset(ctx context.Context, repositoryID RepositoryID, branc
 		return err
 	}
 
-	g.dropTokens(ctx, tokensToDrop)
+	g.dropTokens(ctx, tokensToDrop...)
+	return nil
+}
+
+// resetKey resets given key on branch
+// Since we cannot (will not) modify sealed tokens data, we overwrite changes done on entry on a new staging token, effectively reverting it
+// to the current state in the branch committed data. If entry is not committed return an error
+func (g *KVGraveler) resetKey(ctx context.Context, repositoryID RepositoryID, branch *Branch, key Key, stagedValue *Value, st StagingToken) error {
+	isCommitted := true
+	committed, err := g.Get(ctx, repositoryID, branch.CommitID.Ref(), key)
+	if err != nil {
+		if !errors.Is(err, ErrNotFound) {
+			return err
+		}
+		isCommitted = false
+	}
+
+	if isCommitted { // entry committed and changed in staging area => override with entry from commit
+		if reflect.DeepEqual(committed, stagedValue) {
```

Comment:
This has really bad performance. There's an `Identity` property exactly for that.. I think we already have a `Compare`/`Equal` method for that.